python/lsst/faro/measurement/DetectorTableMeasurement.py

```diff
@@ -0,0 +1,144 @@
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import lsst.pipe.base as pipeBase
+from lsst.verify.tasks import MetricConnections
+import lsst.pex.config as pexConfig
+import lsst.geom
+
+from lsst.faro.base.CatalogMeasurementBase import CatalogMeasurementBaseTaskConfig, CatalogMeasurementBaseTask
+from lsst.pipe.tasks.loadReferenceCatalog import LoadReferenceCatalogTask
+
+__all__ = ("DetectorTableMeasurementConfig", "DetectorTableMeasurementTask")
+
+
+class DetectorTableMeasurementConnections(MetricConnections,
+                                          dimensions=("instrument", "visit", "detector", "band"),
+                                          defaultTemplates={"refDataset": "gaia_dr2_20200414"}):
+
+    catalog = pipeBase.connectionTypes.Input(
+        doc="Source table in parquet format, per visit",
+        dimensions=("instrument", "visit", "band"),
+        storageClass="DataFrame",
+        name="sourceTable_visit",
+        deferLoad=True
+    )
+
+    refcat = pipeBase.connectionTypes.PrerequisiteInput(
+        doc="Reference catalog",
+        name="{refDataset}",
+        storageClass="SimpleCatalog",
+        dimensions=("skypix",),
+        deferLoad=True,
+        multiple=True
+    )
+
+    measurement = pipeBase.connectionTypes.Output(
+        doc="Per-detector measurement",
+        dimensions=("instrument", "visit", "detector", "band"),
+        storageClass="MetricValue",
+        name="metricvalue_{package}_{metric}"
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        if config.connections.refDataset == '':
+            self.prerequisiteInputs.remove("refcat")
+
+
+class DetectorTableMeasurementConfig(CatalogMeasurementBaseTaskConfig,
+                                     pipelineConnections=DetectorTableMeasurementConnections):
+    """Configuration for DetectorTableMeasurementTask."""
+
+    columns = pexConfig.ListField(doc="Columns from sourceTable_visit to load.",
+                                  dtype=str, default=['coord_ra', 'coord_dec', 'detector'])
+
+    refObjLoader = pexConfig.ConfigurableField(
```

Comment:
Now that I've realized that this is not a `LoadReferenceObjectsTask` class, lets not call it `refObjLoader`, since it behaves rather differently (returning a numpy object). Call it `referenceCatalogLoader` maybe?

Reply:
I like this suggestion. Done.